src/controller/debtor-controller.ts

```diff
@@ -0,0 +1,233 @@
+import DebtorService from '../service/debtor-service';
+import User from '../entity/user/user';
+import { asArrayOfUserTypes, asDate } from '../helpers/validators';
+import { In } from 'typeorm';
+import { HandoutFinesRequest } from './request/debtor-request';
+import Fine from '../entity/fine/fine';
+
+export default class DebtorController extends BaseController {
+  private logger: Logger = log4js.getLogger(' DebtorController');
+
+  public constructor(options: BaseControllerOptions) {
+    super(options);
+    this.logger.level = process.env.LOG_LEVEL;
+  }
+
+  public getPolicy(): Policy {
+    return {
+      '/': {
+        GET: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'get', 'all', 'Fine', ['*']),
+          handler: this.returnAllFineHandoutEvents.bind(this),
+        },
+      },
+      '/:id(\\d+)': {
+        GET: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'get', 'all', 'Fine', ['*']),
+          handler: this.returnSingleFineHandoutEvent.bind(this),
+        },
+      },
+      '/single/:id(\\d+)': {
+        DELETE: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'delete', 'all', 'Fine', ['*']),
+          handler: this.deleteFine.bind(this),
+        },
+      },
+      '/eligible': {
+        GET: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'get', 'all', 'Fine', ['*']),
+          handler: this.calculateFines.bind(this),
+        },
+      },
+      '/handout': {
+        POST: {
+          policy: async (req) => this.roleManager.can(req.token.roles, 'get', 'all', 'Fine', ['*']),
+          handler: this.handoutFines.bind(this),
+        },
+      },
+    };
+  }
+
+  /**
+   * Get all fine handout events
+   * @route GET /fines
+   * @group debtors - Operations of the debtor controller
+   * @security JWT
+   * @param {integer} take.query - How many entries the endpoint should return
+   * @param {integer} skip.query - How many entries should be skipped (for pagination)
+   * @returns {PaginatedFineHandoutEventResponse.model} 200 - All existing fine handout events
+   * @returns {string} 400 - Validation error
+   * @returns {string} 500 - Internal server error
+   */
+  public async returnAllFineHandoutEvents(req: RequestWithToken, res: Response): Promise<void> {
+    this.logger.trace('Get all fine handout events by ', req.token.user);
+
+    let take;
+    let skip;
+    try {
+      const pagination = parseRequestPagination(req);
+      take = pagination.take;
+      skip = pagination.skip;
+    } catch (e) {
+      res.status(400).send(e.message);
+      return;
+    }
+
+    try {
+      res.json(await DebtorService.getFineHandoutEvents({ take, skip }));
+    } catch (error) {
+      this.logger.error('Could not return all fine handout event:', error);
+      res.status(500).json('Internal server error.');
+    }
+  }
+
+  /**
+   * Get all fine handout events
+   * @route GET /fines/{id}
```

Comment:
mist unieke @operationId

Reply:
Resolved in cfcbd191b85b3707a4b3e2321c59065dc4a95ea6